edgelet/edgelet-core/src/virtualization.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use crate::error::{Error, ErrorKind};
+use failure::ResultExt;
+use std::process::Command;
+
+pub fn is_virtualized_env() -> Result<Option<bool>, Error> {
+    if cfg!(target_os = "linux") {
+        let status = Command::new("systemd-detect-virt")
+            .status()
+            .context(ErrorKind::GetVirtualizationStatus)?;
+
+        match status.code() {
```

Comment:
I would have used `status.code().and_then(|s| s == 0)`, but a match works perfectly well. You could also do an `if let`.